pkg/client/rules.go

```diff
@@ -76,11 +104,18 @@ func (r *CortexClient) GetRuleGroup(ctx context.Context, namespace, groupName st
 
 // ListRules retrieves a rule group
 func (r *CortexClient) ListRules(ctx context.Context, namespace string) (map[string][]rwrulefmt.RuleGroup, error) {
-	path := "/api/prom/rules"
+	path := rulerAPIPath
+	if r.legacy {
+		path = legacyAPIPath
+	}
 	if namespace != "" {
```

Comment:
This can be simplified if the path prefix is stored as a string in the client struct.

Reply:
I agree. I'll simplify it.